eZ/Bundle/EzPublishCoreBundle/EventListener/ViewControllerListener.php

```diff
@@ -72,77 +69,23 @@ public static function getSubscribedEvents()
     public function getController(FilterControllerEvent $event)
     {
         $request = $event->getRequest();
-        // Only taking content related controller (i.e. ez_content:viewLocation or ez_content:viewContent)
-        if (strpos($request->attributes->get('_controller'), 'ez_content:') === false) {
-            return;
-        }
-        try {
-            if ($request->attributes->get('locationId')) {
-                $valueObject = $this->repository->getLocationService()->loadLocation(
-                    $request->attributes->get('locationId')
-                );
-                $request->attributes->set('contentId', $valueObject->contentId);
-                $request->attributes->set('location', $valueObject);
-            } elseif (($location = $request->attributes->get('location')) instanceof Location) {
-                $valueObject = $location;
-                $request->attributes->set('locationId', $valueObject->id);
-                $request->attributes->set('contentId', $valueObject->contentId);
-            } elseif ($request->attributes->has('contentId')) {
-                $valueObject = $this->repository->sudo(
-                    function (Repository $repository) use ($request) {
-                        return $repository->getContentService()->loadContentInfo(
-                            $request->attributes->get('contentId')
-                        );
-                    }
-                );
-            } elseif (($contentInfo = $request->attributes->get('contentInfo')) instanceof ContentInfo) {
-                $valueObject = $contentInfo;
-                $request->attributes->set('contentId', $valueObject->id);
-            }
-        } catch (UnauthorizedException $e) {
-            throw new AccessDeniedException();
-        }
-
-        if (!isset($valueObject)) {
-            $this->logger->error('Could not resolver a view controller, invalid value object to match.');
 
+        if (($viewBuilder = $this->viewBuilderRegistry->get($request->attributes->get('_controller'))) === null) {
```

Comment:
Hum, why would we do that ? It doesn't return a controller, it builds a View object.